Meme/Meme/MemeTemplatesViewController.swift

```diff
@@ -24,8 +24,31 @@ class MemeTemplatesViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        self.viewModel.requestMemes { [weak self] memes in
-            self?.memes = memes
+        self.viewModel.requestTemplates { [weak self] result in
+            switch result {
+            case .success(let templates):
+                self?.templates = templates
+            case .failure(let error):
+                switch error {
+                case .jsonDeserializationError:
```

Comment:
Let's clean up this callback by handling the error in its own method. Optionally we could use a coordinator object that returns a UIAlertController customized based on the Error passed in. That makes the error handling logic reusable elsewhere.

Reply:
Also this function (presentAlert) should be dispatched back to the main thread.

Reply:
could even be an extension on UIAlertController

Reply:
Sounds good  👌 👍